src/manage_sql_secinfo.c

```diff
@@ -1995,6 +2000,137 @@ update_cert_bund_advisories (int last_cert_update)
 
 /* SCAP update: CPEs. */
 
+/**
+ * @brief Buffer for INSERT statements.
+ */
+typedef struct
+{
+  array_t *statements;     ///< Buffered statements.
+  GString *statement;      ///< Current statemet.
+  int current_chunk_size;  ///< Number of rows in current statement.
+  int max_chunk_size;      ///< Max number of rows per INSERT.
+} inserts_t;
+
+/**
+ * @brief Check size of current statement.
+ *
+ * @param[in]  inserts         Insert buffer.
+ * @param[in]  max_chunk_size  Max chunk size.
+ *
+ * @return Whether this is the first value in the statement.
+ */
+static void
+inserts_init (inserts_t *inserts, int max_chunk_size)
+{
+  inserts->statements = make_array ();
+  inserts->statement = NULL;
+  inserts->current_chunk_size = 0;
+  inserts->max_chunk_size = max_chunk_size;
+}
+
+/**
+ * @brief Close the current statement.
+ *
+ * @param[in]  inserts  Insert buffer.
+ */
+static void
+inserts_statement_close (inserts_t *inserts)
+{
+  if (inserts->statement)
+    g_string_append
+     (inserts->statement,
+      " ON CONFLICT (uuid) DO UPDATE"
+      " SET name = EXCLUDED.name,"
+      "     title = EXCLUDED.title,"
+      "     creation_time = EXCLUDED.creation_time,"
+      "     modification_time = EXCLUDED.modification_time,"
+      "     status = EXCLUDED.status,"
+      "     deprecated_by_id = EXCLUDED.deprecated_by_id,"
+      "     nvd_id = EXCLUDED.nvd_id");
+}
+
+/**
+ * @brief Check size of current statement.
+ *
+ * @param[in]  inserts  Insert buffer.
+ *
+ * @return Whether this is the first value in the statement.
+ */
+static int
+inserts_check_size (inserts_t *inserts)
```

Comment:
As with `inserts_statement_close`, the function name should reflect that this is for CPEs or have a way to define the start of the new statement elsewhere. (I think the struct would be a good place to ensure it's consistent.)

Reply:
Yes, good idea.  I was just trying to be quick.  I've extended inserts_init so the caller can specify the wrapper SQL.